### PR TITLE
work around squire duplicating newlines in copied content

### LINKED ISSUE #5912: empty lines are duplicated when copied from composer
**Bug in web app and desktop app**

empty lines are duplicated when copied from composer

**To Reproduce**
1. begin composing a new message or editing any other HTML (event description, notification email, out of office notification)
2. type a message containing empty lines.

image

3. copy the message to a text editor

**Actual Behavior**
The empty lines in the message are duplicated.

image

**Expected behavior**
The copied message is identical to the original.

image

**Desktop (please complete the following information):**
 - OS: Fedora Linux 37 (Xfce) x86_64
 - Browser: Mullvad Browser
 - Version: 12.5.4
 
**Additional context**
- Tutanota Desktop Version: v3.118.8

- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

### Test Notes 
* test this in firefox, chrome, safari, desktop, both mobile platforms
* also try selection only parts of the editor contents before copying/pasting
* [ ] the bug does not reproduce anymore
* [ ] all the formatting functions in the rich text editor still work
* [ ] inline images are still attached correctly
* [ ] numbered and unnumbered lists are still created after typing `1. ` or `* ` at the start of a line
* [ ] copying and pasting HTML from/to the editor works as expected

### PR DESCRIPTION
**Pull request by ganthern**
squire by default duplicates newlines in the plaintext data when copying content from the editor, but squire.getSelectedText() contains the correct number of newlines.

there's also another entry in the data that contains "text/html" which is not affected by this.

close #5912

### CODE REVIEW

**Comment by ganthern** [comment]
Still need test notes, but I'm mainly wondering if this approach is useful at all. We probably also want to fix this in squire itself, there's one or two issues about weird whitespace / `<br/>` on their repo.

Note that not being able to copy/paste inline images is unrelated to this.

File: src/gui/editor/Editor.ts [hunk 1]
```diff
@@ -60,6 +60,14 @@ export class Editor implements ImageHandler, Component {
 	 * * "input" event on squire resets flag.
 	 */
 	private pasteListener: (e: ClipboardEvent) => void = (_: ClipboardEvent) => (this.userHasPasted = true)
+	private copyListener: (e: ClipboardEvent) => void = (e: ClipboardEvent) => {
+		if (e.clipboardData == null) return
+		// squire by default duplicates newlines when copying content from the editor, but .getSelectedText()
+		// contains the correct number of \n
+		// there's also another entry in the data that contains the "text/html" which is not affected by this.
+		if (!e.clipboardData.types.includes("text/plain")) return
+		e.clipboardData.setData("text/plain", this.squire.getSelectedText())
```
Comment (charlag) [inline src/gui/editor/Editor.ts:69]: is `getSelectedText` always plaintext?

**Review by charlag — COMMENTED** [review]
Like you said, seems a bit pointless and should probably be fixed in Squire (or maybe we should move to another editor)

File: src/gui/editor/Editor.ts (hunk 1 above)
Comment (ganthern) [reply src/gui/editor/Editor.ts:69]: squire only walks text nodes and appends their data field (which is plain text) to an accumulator

**Comment by ganthern** [comment]
I've raised the issue with the squire maintainers https://github.com/fastmail/Squire/issues/440

There's obviously no timeline on a fixed version being released if we go the route of fixing it directly at the source.

**Comment by ganthern** [comment]
squire fixed the bug as of 2.2.2

**Pull request closed without merging**